### PR TITLE
Clean up mouse region code

### PR DESCRIPTION
**Pull request by chunhtai**
This feels like a dead code, the _validForMouseTracker is set to false in RenderMouseRegion.detach. If the element.mounted = false, the RenderMouseRegion must already be detached
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by dkwingsmt — APPROVED** [review]
Thanks for finding this! The `_validForMouseTracker` logic, which was added in https://github.com/flutter/flutter/pull/69866, overshadowed this part of logic. Also, the behavior that this part of logic was for is tested in "doesn't trigger pointer exit when widget disappears", so having no tests broken means that this change is safe.

**Comment by Hixie** [comment]
test-exempt: refactor with no semantic change

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -6172,7 +6172,7 @@ class Listener extends SingleChildRenderObjectWidget {
 ///
 ///  * [Listener], a similar widget that tracks pointer events when the pointer
 ///    has buttons pressed.
-class MouseRegion extends StatefulWidget {
+class MouseRegion extends StatelessWidget {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart:6175]: I didn't look closely but is it possible that we could remove the _RawMouseRegion entirely? Or rather, remove MouseRegion entirely, and rename _RawMouseRegion to MouseRegion?

It's not clear what purpose there is in having them separate now.

Comment (chunhtai) [reply packages/flutter/lib/src/widgets/basic.dart:6175]: ah yes, what was I thinking about. lol. will do that

**Comment by chunhtai** [comment]
@Hixie done

**Review by jambestwick — COMMENTED** [review]
vi

**Pull request merged**